src/main/java/com/minecolonies/client/gui/WindowHutGuardTower.java

```diff
@@ -0,0 +1,467 @@
+     */
+    private static final String PAGE_ACTIONS    = "levelActions";
+
+    /**
+     * Id of the previous page button in the GUI
+     */
+    private static final String BUTTON_PREVPAGE = "prevPage";
+
+    /**
+     * Id of the next page button in the GUI.
+     */
+    private static final String BUTTON_NEXTPAGE = "nextPage";
+
+    /**
+     * Id of the switch job button in the GUI.
+     */
+    private static final String BUTTON_JOB = "job";
+
+    /**
+     * Id of the switch assignment mode button in the GUI - (Manually / Automatically).
+     */
+    private static final String BUTTON_ASSIGNMENT_MODE = "assign";
+
+    /**
+     * Id of the switch patrolling mode button in the GUI - (Manually / Automatically).
+     */
+    private static final String BUTTON_PATROL_MODE = "patrol";
+
+    /**
+     * Id of the switch retrieval mode button in the GUI - (Off / 10% / 20%).
+     */
+    private static final String BUTTON_RETRIEVAL_MODE = "retrieve";
+
+    /**
+     * Id of the switch the task button in the GUI - (Patrol).
+     */
+    private static final String BUTTON_TASK_PATROL= "patrolling";
+
+    /**
+     * Id of the switch the task button in the GUI - (Follow).
+     */
+    private static final String BUTTON_TASK_FOLLOW = "following";
+
+    /**
+     * Id of the switch the task button in the GUI - (Guard).
+     */
+    private static final String BUTTON_TASK_GUARD = "guarding";
+
+    /**
+     * Id of the settarget button in the GUI - Depending on task sets guard position or patrol..
+     */
+    private static final String BUTTON_SET_TARGET = "setTarget";
+
+
+    private static final String VIEW_PAGES                      = "pages";
+    private static final String HUT_GUARD_TOWER_RESOURCE_SUFFIX = ":gui/windowHutGuardTower.xml";
+    private Button                  buttonPrevPage;
+    private Button                  buttonNextPage;
+
+    /**
+     * Assign the job manually, knight or ranger.
+     */
+    private boolean assignManually = false;
+
+    /**
+     * Retrieve the guard on low health.
+     */
+    private boolean retrieveOnLowHealth = false;
+
+    /**
+     * Patrol manually or automatically.
+     */
+    private boolean patrolManually = false;
+
+    /**
+     * The task of the guard, following the Task enum.
+     */
+    private BuildingGuardTower.Task task = BuildingGuardTower.Task.GUARD;
+
+    /**
+     * The job of the guard, following the GuarJob enum.
+     */
+    private BuildingGuardTower.GuardJob job = null;
+
+    /**
+     * The list of manual patrol targets.
+     */
+    private ArrayList<BlockPos> patrolTargets = new ArrayList<>();
+
+    /**
+     * The patrol list.
+     */
+    private ScrollingList patrolList;
+
+    /**
+     * Buttons used in the application:
+     */
+    private final Button buttonTaskPatrol;
+    private final Button buttonTaskFollow;
+    private final Button buttonTaskGuard;
+    private final Button buttonSetTarget;
+
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)